### PR TITLE
chore(NODE-6426): update prod dependencies

### PR DESCRIPTION
**Pull request by durran**
### Description

Updates prod dependencies

#### What is changing?

Updates the prod dependencies to latest semver compatible versions.

##### Is there new documentation needed for these changes?

None

#### What is the motivation for this change?

NODE-6426

### Double check the following

- [x] Ran `npm run check:lint` script
- [x] Self-review completed using the [steps outlined here](https://github.com/mongodb/node-mongodb-native/blob/HEAD/CONTRIBUTING.md#reviewer-guidelines)
- [x] PR title follows the [correct format](https://www.conventionalcommits.org/en/v1.0.0/): `type(NODE-xxxx)[!]: description`
  - Example: `feat(NODE-1234)!: rewriting everything in coffeescript`
- [ ] Changes are covered by tests
- [ ] New TODOs have a related JIRA ticket

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -25,15 +25,15 @@
...
-    "@mongodb-js/saslprep": "^1.1.5",
-    "bson": "^6.7.0",
+    "@mongodb-js/saslprep": "^1.1.9",
+    "bson": "^6.9.0",
     "mongodb-connection-string-url": "^3.0.0"
   },
   "peerDependencies": {
-    "@aws-sdk/credential-providers": "^3.188.0",
-    "@mongodb-js/zstd": "^1.1.0",
+    "@aws-sdk/credential-providers": "^3.687.0",
```
[the remark this answers is not in the document]
Comment (durran) [reply package.json:33]: Oh good catch, I forgot about that. I will check against the runtime and update appropriately.

[the remark this answers is not in the document]
Comment (durran) [reply package.json]: Updated, confirmed runtime SDK is now 3.632.0

[the remark this answers is not in the document]
Comment (nbbeeken) [reply package.json]: Sorry if I was unclear, that was the reason in the past that we actually downgraded the AWS SDK but we should still treat AWS the same as the other peers, it cannot be upgraded until we release a major version, or we change our policy such that we can upgrade any of these.

**Review by nbbeeken — CHANGES_REQUESTED** [review]
for viz

File: package.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (durran) [reply package.json]: My mistake - I misunderstood the comment. Reverted the change.

**Review by nbbeeken — CHANGES_REQUESTED** [review]
Something is failing relating to snappy?

**Comment by durran** [comment]
New PR: https://github.com/mongodb/node-mongodb-native/pull/4327

**Pull request closed without merging**